### ISSUE TITLE
How to use "calc-grab-region" without pre-selection?

### ISSUE THREAD

**Issue by salutis** [oantolin/embark#378]
Say I have the following mathematical expression in a buffer:

1+2

To evaluate it, I move the point before `1`, then press `C-SPC`, then move the point after `2`, then press `C-.` to call Embark, and then press `*` to calculate the expression in the region. Why cannot type `C-.` on the expression and then `*` to evaluate it?

Curiously, when I type `C-.` on the expression without pre-selecting it, Embark provides, for example "Delete region", and it deletes the expression. Why does that menu exinclude "Calculate region" too? Or does the word "region" mean something else when I select the region myself?

**Comment by oantolin** [comment]
> Or does the word "region" mean something else when I select the region myself?

I just realized I forgot to address this. Embark uses region in the same sense as Emacs: the part of the buffer between the point and the mark. When you use  embark-act, Embark highlights the target, but that highlighting is not a region. If you want to use a region command you have to have explicitly marked a region, or use Embark's `mark` action bound to `SPC`, or have configured an action to use automarking as above

**Comment by salutis** [comment]
Thank you _so much_ for taking time to explain the matter to me; I found your answer super-useful. As a 101 user, I find the "automark" functionality confusing, but I also understand its appeal. Thank you again, @oantolin!

**Comment by woolsweater** [comment]
> When you use embark-act without a region being active ...

This is a super tip that also answered my question about another `(interactive "r")` command that I wanted to use with Embark. It seems like it would fit well on the wiki. Do you mind if I add it there, or would you prefer not since `embark--mark-target` is private?

**Comment by oantolin** [comment]
What command is it? The double hyphen in `embark--mark-target` is not an obstacle. I do document those in the manual, the double hyphen means more "advanced" than "private" in their case. (I still reserve the right to change them without warning, though 😛).

**Comment by woolsweater** [comment]
Oh, it was [a custom command of my own](https://gitlab.com/woolsweater/dotemacs.d/-/blob/cde20a161068f56cfeba40b8c8f285e7b6301f18/startup/builtin.el#L314-325) to imitate some native Mac behavior.

Sounds good, thanks!

**Comment by oantolin** [comment]
That's an interesting command! When do you use it instead of immediately starting a search? (Note that for immediately searching Embark already has an embark-isearch general action, and embark-consult also adds all the consult search commands as general actions.)

**Comment by woolsweater** [comment]
Yeah, absolutely. To be honest it's mostly a "muscle memory" command -- I just want ⌘ E to do what I expect from long years of Mac use. In actual practice I use `M-s .` and `C-. C-s` much more often. 😄

**Comment by oantolin** [comment]
Well, please feel free to add it to the wiki, mentioning that it gets you the same ⌘ E workflow you were used to.
